pandas/core/arrays/datetimelike.py

```diff
@@ -552,7 +552,12 @@ def _concat_same_type(
 
             if obj.freq is not None and all(x.freq == obj.freq for x in to_concat):
                 pairs = zip(to_concat[:-1], to_concat[1:])
-                if all(pair[0][-1] + obj.freq == pair[1][0] for pair in pairs):
+                # error: No overload variant of "__radd__" of "BaseOffset" matches
+                # argument type "NaTType"
```

Comment:
is this something that should be fixed in the BaseOffset .pyi file?  (OK if you want to leave that as out of scope)

Reply:
I added `NaTType` for just `__radd__`. Would need to add it to many more oprators.